rapid7_vulndb/komand_rapid7_vulndb/util/utils.py

```diff
@@ -0,0 +1,311 @@
+from komand.exceptions import PluginException
+
+from typing import Dict, List, Callable, Tuple
+import requests
+import copy
+import sys
+from urllib.parse import urljoin
+
+
+class R7VDB:
+    """
+    Rapid7 vulnerability database API utilities
+    This class is used by the rapid7_vulndb plugin for
+    extracting https://vdb.rapid7.com/swagger_doc API data
+
+    Two main functions that are used by the plugin are
+        1. paginate_search
+        2. make_content_result
+
+    most of the methods implemented as class methods to avoid initialization
+    """
+
+    search_url: str = "https://vdb-kasf1i23nr1kl2j4.rapid7.com/v1/search"
+    content_url: str = 'https://vdb-kasf1i23nr1kl2j4.rapid7.com/v1/content/'
+    seach_fields: List[str] = ["title", "published_at", "identifier"]
+    content_common_fields: List[str] = ["title", "description", "content_type",
+                                        "published_at", "references"]
+    content_module_fields: List[str] = ["architectures", "authors", "rank",
+                                        "reliability"]
+    content_vuln_fields: List[str] = ["severity", "solutions", "alternate_ids"]
+    content_module_fields_to_serialize: List[str] = ["architectures",
+                                                     "references", "authors"]
+    content_vuln_fields_to_serialize: List[str] = ["solutions", "references"]
+
+    @classmethod
+    def paginate_search(cls, query: Dict, num_of_pages: int) -> List[Dict]:
+        fn = sys._getframe().f_code.co_name
```

Comment:
sys._getframe() should not be used - it is a private function in `sys` and may be unsafe to use between Python versions.

From the docs:
```
This function should be used for internal and specialized purposes only.
```